neptune/internal/backends/hosted_neptune_backend.py

```diff
@@ -315,20 +315,29 @@ def create_notebook(self, project):
             raise ProjectNotFound(project_identifier=project.full_id)
 
     @with_api_exceptions_handler
-    def create_checkpoint(self, notebook_id, jupyter_path, _file):
-        with self._upload_raw_data(api_method=self.leaderboard_swagger_client.api.createCheckpoint,
-                                   data=_file,
-                                   headers={"Content-Type": "application/octet-stream"},
-                                   path_params={
-                                       "notebookId": notebook_id
-                                   },
-                                   query_params={
-                                       "jupyterPath": jupyter_path
-                                   }) as response:
-            if response.status_code == NOT_FOUND:
-                raise NotebookNotFound(notebook_id=notebook_id)
-            else:
-                response.raise_for_status()
+    def create_checkpoint(self, notebook_id, jupyter_path, _file=None):
+        if _file is not None:
+            with self._upload_raw_data(api_method=self.leaderboard_swagger_client.api.createCheckpoint,
+                                       data=_file,
+                                       headers={"Content-Type": "application/octet-stream"},
+                                       path_params={
+                                           "notebookId": notebook_id
+                                       },
+                                       query_params={
+                                           "jupyterPath": jupyter_path
+                                       }) as response:
+                if response.status_code == NOT_FOUND:
+                    raise NotebookNotFound(notebook_id=notebook_id)
+                else:
+                    response.raise_for_status()
+                    CheckpointDTO = self.leaderboard_swagger_client.get_model('CheckpointDTO')
+                    return CheckpointDTO.unmarshal(response.json())
+        else:
+            NewCheckpointDTO = self.leaderboard_swagger_client.get_model('NewCheckpointDTO')
+            return self.leaderboard_swagger_client.api.createCheckpoint2(
```

Comment:
Can we please rename this to something more meaningful than `..2`? `createIncorporealCheckpoint` (as in "no body") maybe?

Reply:
I'm actually wondering if we shouldn't leave the old method there as well for the long run..